### ISSUE TITLE
Rule no-unnecessary-type-assertion doesn't work if tsconfig has `"strict": true`

### ISSUE THREAD

**Issue by guidsdo** [palantir/tslint#4840]
### Bug Report
Rule no-unnecessary-type-assertion doesn't work if your `tsconfig.json` has `"strict": true`. It was made so that it only works if you have `"strictNullChecks": true` but should also work for `"strict": true` since that covers all strict rules.

Bug introduction: https://github.com/palantir/tslint/commit/b836f285e63d7dc44908eae40ef5d6feb16e026a#diff-2237d1e758ab18614b7d976b89e56ea0R53
Strict rule explanation: https://www.typescriptlang.org/docs/handbook/compiler-options.html

Note, will fix this myself so PR incoming.

#### Reproduction using [TSLint Playground](https://palantir.github.io/tslint-playground)
Playground doesn't allow you to change the tsconfig?

#### TypeScript code being linted

_tsconfig.json_
```json
{
  "compilerOptions": {
    /* Enable all strict type-checking options. */
    "strict": true,
    /* Enable strict null checks. */
    // "strictNullChecks": true
  }
}
```

_Code_
```ts
const x = "y";
x!.length;
```

_with `tslint.json` configuration:_
```json
{
  "defaultSeverity": "error",
  "rules": {
    "no-unnecessary-type-assertion": true
  }
}
```

#### Actual behavior
No errors.

#### Expected behavior
An error!

**Comment by JoshuaKGoldberg** [comment]
🤖 Beep boop! 👉  **TSLint is deprecated** 👈  and you should **switch to [typescript-eslint](https://typescript-eslint.io)**! 🤖 

🔒 This issue is being locked to prevent further unnecessary discussions. Thank you! 👋
